### DISCUSSION TITLE
shairport-sync that start from systemctl will always choose card o even alsaconf told not so

### DISCUSSION THREAD

**Discussion by teelekkung** [mikebrady/shairport-sync#1715] [Q&A]
I have setup my rpi4b to  Shairport-sync (fresh Pi os lite kernel 6.1)
with [this guide](https://github.com/mikebrady/shairport-sync/blob/master/CAR%20INSTALL.md)
everything go well until I introduce usb_dac(fifo btr5) to system
pi detect sound card fine , set `raspi-config` to out to sound card, `speaker-test -c 2 -t wav `

Every thing seem fine , after that `shairport-sync` to manual start and you know what, sound output trought my dac

so I go to rc.local as guide to start when boot and~~~~    it out put to card 0 (headphone)
after some fussing I revert rc.local to default then try `systemctl start shairport-sync`. Samething Happened 

I know that reodering from /etc/modprobe.d/alsa-base.conf will fix just fine but the problem is when unplug dac, Shairport-sync will refuse to start because of can't find output (safe protocol somethingๆ) (too lazy to reproduce that error output)

<details><summary>Output From speaker-test</summary>
<p>

```bash
speaker-test 1.2.4
Playback device is default
Stream parameters are 48000Hz, S16_LE, 2 channels
WAV file(s)
Rate set to 48000Hz (requested 48000Hz)
Buffer size range from 32 to 96000
Period size range from 16 to 48000
Using max buffer size 96000
Periods = 4
was set period_size = 24000
was set buffer_size = 96000
 0 - Front Left
 1 - Front Right
Time per period = 1.518432
 0 - Front Left
 1 - Front Right
Time per period = 3.000092
 0 - Front Left
 1 - Front Right
```

</p>
</details>

<details><summary>Output From shairport-sync -X</summary>
<p>

```bash
>> Display Config Start.

From "uname -a":
 Linux raspi 6.1.21-v8+ #1642 SMP PREEMPT Mon Apr  3 17:24:16 BST 2023 aarch64 GNU/Linux

From /etc/os-release:
 Debian GNU/Linux 11 (bullseye)

From /sys/firmware/devicetree/base/model:
 Raspberry Pi 4 Model B Rev 1.1

Shairport Sync Version String:
 4.2.1d0-55-g8821ec28-AirPlay2-smi9-alac-OpenSSL-Avahi-ALSA-soxr-metadata-sysconfdir:/etc

Command Line:
 shairport-sync -X

Configuration File:
 /etc/shairport-sync.conf

Configuration File Settings:
 general : 
 {
   name = "Sahakiyat's HiFi Speaker";
 };

>> Display Config End.
>> Goodbye!
```

</p>
</details>

<details><summary>My .asoundrc</summary>
<p>

```bash
pcm.!default {
  type asym
  playback.pcm {
    type plug
    slave.pcm "output"
  }
  capture.pcm {
    type plug
    slave.pcm "input"
  }
}

pcm.output {
  type hw
  card 3
}

ctl.!default {
  type hw
  card 3
}
```

</p>
</details>

<details><summary>my aplay -l</summary>
<p>

```bash
**** List of PLAYBACK Hardware Devices ****
card 0: Headphones [bcm2835 Headphones], device 0: bcm2835 Headphones [bcm2835 Headphones]
  Subdevices: 8/8
  Subdevice #0: subdevice #0
  Subdevice #1: subdevice #1
  Subdevice #2: subdevice #2
  Subdevice #3: subdevice #3
  Subdevice #4: subdevice #4
  Subdevice #5: subdevice #5
  Subdevice #6: subdevice #6
  Subdevice #7: subdevice #7
card 1: vc4hdmi0 [vc4-hdmi-0], device 0: MAI PCM i2s-hifi-0 [MAI PCM i2s-hifi-0]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 2: vc4hdmi1 [vc4-hdmi-1], device 0: MAI PCM i2s-hifi-0 [MAI PCM i2s-hifi-0]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 3: BTR5 [FiiO BTR5], device 0: USB Audio [USB Audio]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
```

</p>
</details>

**Comment by mikebrady** [comment]
Thanks for this, but this is really an Issue rather than a Discussion item. Perhaps you would open an Issue about it.
